### ISSUE TITLE
ROL build failure for CUDA 8.0 Debug build on white/ride

### ISSUE THREAD

**Issue by bartlettroscoe** [trilinos/Trilinos#3072]
CC: @trilinos/rol, @rppawlo (Trilinos Nonlinear Solvers Product Lead)

## Description

The compilation of the file `packages/rol/example/burgers-control/example_01.cpp` fails in the CUDA 8.0 debug build `Trilinos-atdm-white-ride-cuda-debug-pt-all-at-once` on 'white' and 'ride'.  The build error output for the build this morning shown [here](https://testing-vm.sandia.gov/cdash/viewBuildError.php?buildid=3692122) on 'white' shows:

```
...

/home/rabartl/WHITE/ATDM_Driver/Trilinos-atdm-white-ride-cuda-debug-pt-all-at-once/SRC_AND_BUILD/Trilinos/packages/rol/example/burgers-control/example_01.hpp(522): error: the type in a dynamic_cast must be a pointer or reference to a complete class type, or void *

...

1 error detected in the compilation of "/tmp/tmpxft_00001203_00000000-5_example_01.cpp4.ii".
```

## Steps to reproduce

This build error can reproduced on 'white' or 'ride' as described in the document:

* https://github.com/trilinos/Trilinos/blob/develop/cmake/std/atdm/README.md

The specific instructions for 'white' or 'ride' are given at:

* https://github.com/trilinos/Trilinos/blob/develop/cmake/std/atdm/README.md#ridewhite

The one difference is that this build of all of the Primary Tested Trilinos packages (that includes more package than are being used by ATDM APPs currently) does not exclude any Trilinos packages and tweaks a few other settings so it uses the file `Trilinos/cmake/std/atdm/ATDMDevEnvAllPtPackages.cmake` instead of the file `ATDMDevEnv.cmake`.

After cloning Trilinos, the following commands should reproduce the build failure:

```
$ cd <some_build_dir>/

$ source $TRILINOS_DIR/cmake/std/atdm/load-env.sh cuda-debug

$ cmake \
  -GNinja \
  -DTrilinos_CONFIGURE_OPTIONS_FILE:STRING=cmake/std/atdm/ATDMDevEnvAllPtPackages.cmake \
  -DTrilinos_ENABLE_TESTS=ON -DTrilinos_ENABLE_ROL=ON \
  $TRILINOS_DIR

$ make NP=16
```

I (@bartlettroscoe) just tired this on 'white' and I was able to reproduce the same build failure shown on CDash shown above.

**Comment by bartlettroscoe** [comment]
The fix seems to be trivial.  I will fix this in the branch I am working in for #3069 and I will create a PR to fix this.

**Comment by bartlettroscoe** [comment]
With the merge of PR #3081 yesterday, this ROL CUDA build failure is now gone as shown [here](https://testing-vm.sandia.gov/cdash/index.php?project=Trilinos&parentid=3705319).  Note the `-1` subscript on the `0` under the "Build Error" column for ROL.

Closing as complete.
